mithril-common/src/crypto_helper/cardano/key_certification.rs

```diff
@@ -221,18 +221,17 @@ impl KeyRegWrapper {
             opcert
                 .validate()
                 .map_err(|_| ProtocolRegistrationErrorWrapper::OpCertInvalid)?;
-            // TODO: List of eligible indices to be defined by CurrentKesPeriod and StartKesPeriod
             let mut pool_id = None;
             let sig = kes_sig.ok_or(ProtocolRegistrationErrorWrapper::KesSignatureMissing)?;
-            for kes_period_try in 0..64 {
+            let kes_period =
+                kes_period.ok_or(ProtocolRegistrationErrorWrapper::KesPeriodMissing)?;
+            let kes_period_try_min = std::cmp::max(0, kes_period as i32 - 1) as usize;
+            let kes_period_try_max = std::cmp::min(64, kes_period as i32 + 1) as usize;
```

Comment:
Do you need to use i32 here? The original type of kes should work, as we will not result with a negative num.